packages/playwright/src/index.ts

```diff
@@ -701,7 +698,80 @@ class ArtifactsRecorder {
     })));
 
     await this._screenshotRecorder.persistTemporary();
-    await this._pageSnapshotRecorder.persistTemporary();
+    if (this._errorPrompts) {
+      await this._takePageSnapshot();
+      await this._attachErrorPrompts();
+    }
+  }
+
+  private async _takePageSnapshot(context?: BrowserContext) {
```

Comment:
Let's not take the snapshot if `testInfo.errors` is empty.

Reply:
implemented that for the case where there's leftover contexts. if the browser context is closed before the test end, we need to decide if we want to take a snapshot before knowing if there's errors.

Reply:
revisited in https://github.com/microsoft/playwright/pull/34995/commits/08c96873dc8b6fbbaecb09d508eb5c7cd04c53f3